src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.cs

```diff
@@ -1204,8 +1204,51 @@ private TokenValidationResult ValidateJWS(string token, TokenValidationParameter
         {
             try
             {
-                var jsonWebToken = ValidateSignature(token, validationParameters, configuration);
-                return ValidateTokenPayload(jsonWebToken, validationParameters, configuration);
+                JsonWebToken jsonWebToken;
+                TokenValidationResult tokenValidationResult;
+                if (validationParameters.SignatureValidator != null || validationParameters.SignatureValidatorUsingConfiguration != null)
+                {
+                    return ValidateTokenPayload(
```

Comment:
Shouldn't we check ValidateSignatureLast here as well?

Reply:
SignatureValidator and SignatureValidatorUsingConfiguration will return a JsonWebToken to validate the payload. Set tvp.ValidateSignatureLast = false here.